pandas/core/indexes/accessors.py

```diff
@@ -129,9 +129,41 @@ class DatetimeProperties(Properties):
 
     Examples
     --------
-    >>> s.dt.hour
-    >>> s.dt.second
-    >>> s.dt.quarter
+    >>> seconds_series = pd.Series(pd.date_range("2000-01-01", periods=3, freq="s"))
+    >>> seconds_series
+    0   2000-01-01 00:00:00
+    1   2000-01-01 00:00:01
+    2   2000-01-01 00:00:02
+    dtype: datetime64[ns]
+    >>> seconds_series.dt.second
+    0    0
+    1    1
+    2    2
+    dtype: int64
+
+    >>> hours_series = pd.Series(pd.date_range("2000-01-01", periods=3, freq="h"))
+    >>> hours_series
+    0   2000-01-01 00:00:00
+    1   2000-01-01 01:00:00
+    2   2000-01-01 02:00:00
+    dtype: datetime64[ns]
+    >>> hours_series.dt.hour
+    0    0
+    1    1
+    2    2
+    dtype: int64
+
+    >>> quarters_series = pd.Series(pd.date_range("2000-01-01", periods=3, freq="q"))
+    >>> quarters_series
+    0   2000-03-31
+    1   2000-06-30
+    2   2000-09-30
+    dtype: datetime64[ns]
+    >>> quarters_series.dt.quarter
+    0    1
+    1    2
+    2    3
+    dtype: int64
```

Comment:
I wonder whether needs to be in the class description.

with help(pd.core.indexes.accessors.DatetimeProperties)

under Data descriptors defined here:

we have

 |  dayofweek
 |      The day of the week with Monday=0, Sunday=6.
 |      
 |      Return the day of the week. It is assumed the week starts on
 |      Monday, which is denoted by 0 and ends on Sunday which is denoted
 |      by 6. This method is available on both Series with datetime
 |      values (using the `dt` accessor) or DatetimeIndex.
 |      
 |      Returns
 |      -------
 |      Series or Index
 |          Containing integers indicating the day number.
 |      
 |      See Also
 |      --------
 |      Series.dt.dayofweek : Alias.
 |      Series.dt.weekday : Alias.
 |      Series.dt.day_name : Returns the name of the day of the week.
 |      
 |      Examples
 |      --------
 |      >>> s = pd.date_range('2016-12-31', '2017-01-08', freq='D').to_series()
 |      >>> s.dt.dayofweek
 |      2016-12-31    5
 |      2017-01-01    6
 |      2017-01-02    0
 |      2017-01-03    1
 |      2017-01-04    2
 |      2017-01-05    3
 |      2017-01-06    4
 |      2017-01-07    5
 |      2017-01-08    6
 |      Freq: D, dtype: int64

and for hour, second and quarter we just have 


 |  hour
 |      The hours of the datetime.

second
 |      The seconds of the datetime.

|  quarter
 |      The quarter of the date.

 
maybe these examples should be in the descriptor docstrings instead. wdyt?

Reply:
I believe that we should have them for both, because the class's docstring is telling the user/developer that this class is responsible for accessing the fields, and some examples on.

And on top of that we should also supply specific examples for each field accessor, would love to do so in a follow up.